runtime/array.c

```diff
@@ -524,15 +550,13 @@ static value caml_uniform_array_gather(intnat num_arrays,
     for (mlsize_t i = 0; i < num_arrays; i++) {
       /* Here we can do a direct copy since this cannot create old-to-young
          pointers, nor mess up with the incremental major GC.
-         Since the value is not yet published, plain writes are enough.
          Since the [memcpy] implementation does not guarantee that the reads
          are always word-sized, we explicitly perform word-sized atomic
          loads of the relaxed kind in the *source* array to avoid tearing in
-         case of data races on that array. [MM] */
+         case of data races on that array (see #13950). [MM] */
```

Comment:
Nitpick: the comment "Since the [memcpy] implementation..." is now redundant with the documentation of `wo_memcpy`, so I think that part could be removed here.